doc/website/blog: visualizing schemas with entviz (#3279)
ent/ent: doc/website/blog/2023-01-26-visualizing-with-entviz.md

---
title: Quickly visualize your Ent schemas with entviz
author: Rotem Tamir
authorURL: "https://github.com/rotemtam"
authorImageURL: "https://s.gravatar.com/avatar/36b3739951a27d2e37251867b7d44b1a?s=80"
authorTwitter: _rtam
image: "https://entgo.io/images/assets/entviz-v2.png"
---

### TL;DR

To get a public link to a visualization of your Ent schema, run:

```
go run -mod=mod ariga.io/entviz ./path/to/ent/schema 
```

![](https://entgo.io/images/assets/erd/edges-quick-summary.png)

### Visualizing Ent schemas

Ent enables developers to build complex application data models
using [graph semantics](https://en.wikipedia.org/wiki/Graph_theory): instead of defining tables, columns, association
tables and foreign keys, Ent models are simply defined in terms of [Nodes](https://entgo.io/docs/schema-fields)
and [Edges](https://entgo.io/docs/schema-edges):

```go
package schema

import (
	"entgo.io/ent"
	"entgo.io/ent/schema/edge"
)

// User schema.
type User struct {
	ent.Schema
}

// Fields of the user.
func (User) Fields() []ent.Field {
	return []ent.Field{
		// ...
	}
}

// Edges of the user.
func (User) Edges() []ent.Edge {
	return []ent.Edge{
		edge.To("pets", Pet.Type),
	}
}
```

> Defining data models this way has many benefits such as being able to

Reviewer: ```suggestion
Modeling data this way has many benefits such as being able to
```

that extension enabled users to generate simple, local HTML documents containing entity-relationship diagrams describing
an application's Ent schema.

Today, we're happy to share a [super cool tool](https://github.com/ariga/entviz) by the same name created
by [Pedro Henrique (crossworth)](https://github.com/crossworth) which is a completely fresh take on the same problem.
With (the new) entviz you run a simple Go command:

```
go run -mod=mod ariga.io/entviz ./path/to/ent/schema 
```

The tool will analyze your Ent schema and create a visualization on the [Atlas Playground](https://gh.atlasgo.cloud) and
create a shareable, public [link](https://gh.atlasgo.cloud/explore/c3aa3f24) for you:

```
Here is a public link to your schema visualization:
	    https://gh.atlasgo.cloud/explore/c3aa3f24
```

In this link you will be able to see your schema visually as an ERD or textually as either a SQL
or [Atlas HCL](https://atlasgo.io/atlas-schema/sql-resources) document.

### Wrapping up

In this blog post we discussed some scenarios where you might find it useful to quickly get a visualization of your Ent
application's schema, we then showed how creating such visualizations can be achieved

> using [entviz](https://github.com/ariga/entviz). If you like the idea, we will be super happy if you tried it today and

Reviewer: ```suggestion
using [entviz](https://github.com/ariga/entviz). If you like the idea, we'd be super happy if you tried it today and
```
